### DISCUSSION TITLE
Budget allocator across two models?

### DISCUSSION THREAD

**Discussion by alexanderhe88** [facebookexperimental/Robyn#962] [General]
Hi Robyn developers/community, hope all is well and I appreciate your efforts in this fabulous tool. 
Here is my question. I've built 2 models with exactly the same media/organic/contextual variables but different target variables. Let's say the first model is **Amazon sales**, and the second model is **Shopify sales**.

Is there a way for me to run optimization/budget allocation to optimize the overall sales of Amazon + Shopify?

**Comment by gufengzhou** [comment]
Thanks! At the moment this is not possible unfortunately. It's quite a challenging problem to solve because you're having multiple outcomes with respective curves. Also sometimes certain media might only work for certain sales platform, so there might be further constraints.

**Comment by ThouShallNotBeNamed** [comment]
AllocatorCollect_max_response <- robyn_allocator(
InputCollect = InputCollect,
OutputCollect = OutputCollect,
select_model = select_model,
scenario = "max_response",
channel_constr_low = c(0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5),
channel_constr_up = c(5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0),
expected_spend = 16000000, # << CHANGE THIS to your total budget
expected_spend_days = 136 # << CHANGE THIS to your time period
)

plot(AllocatorCollect_max_response)

I am setting an expected_spend, but it does not seem to reflect in my output. In my plot, i see that the robyn_allocator is setting default as the historical spend.

Can someone please help?
